libs/brig-types/src/Brig/Types/User.hs

```diff
@@ -313,37 +383,46 @@ instance ToJSON RichField where
     -- "value": ...}@ is how all other SCIM payloads are formatted, so it's quite possible
     -- that some provisioning agent would support "type" but not "name".
     toJSON u = object
-        [ "type" .= richFieldType u
+        [ "type" .= CI.original (richFieldType u)
         , "value" .= richFieldValue u
         ]
 
 instance FromJSON RichField where
     parseJSON = withObject "RichField" $ \o -> do
         RichField
-            <$> o .: "type"
+            <$> (CI.mk <$> o .: "type")
             <*> o .: "value"
 
--- | Empty rich info, returned for users who don't have rich info set.
-emptyRichInfo :: RichInfo
-emptyRichInfo = RichInfo
-    { richInfoFields = []
-    }
-
 -- | Calculate the length of user-supplied data in 'RichInfo'. Used for enforcing
 -- 'setRichInfoLimit'
 --
+-- This works on @[RichField]@ so it can be reused by @RichInfoAssocList@
+--
 -- NB: we could just calculate the length of JSON-encoded payload, but it is fragile because
 -- if our JSON encoding changes, existing payloads might become unacceptable.
-richInfoSize :: RichInfo -> Int
-richInfoSize (RichInfo fields) =
-    sum [Text.length t + Text.length v | RichField t v <- fields]
+richInfoAssocListSize :: [RichField] -> Int
+richInfoAssocListSize fields = sum [Text.length (CI.original t) + Text.length v | RichField t v <- fields]
```

Comment:
Personally I prefer `map` (or in this case  `foldMap`) over list comprehensions. But it's not a hill I will die on

Reply:
I have no real preference in this case, I will merge it without any change for now. Maybe next time we touch this code, we can refactor :)